docs/source/llms/openai/notebooks/openai-quickstart.ipynb

```diff
@@ -0,0 +1,505 @@
+    "\n",
+    "If you encounter an Exception stating that the `OPENAI_API_KEY` environment variable must be set, consider the following common issues and remedies:\n",
+    "\n",
+    "- **Kernel Restart**: If you're using a Jupyter notebook, make sure to restart the kernel after setting the environment variable. This is necessary for the kernel to recognize changes to environment variables.\n",
+    "- **Correct Profile Script**: On macOS and Linux, ensure you've edited the correct profile script (.bashrc for Bash, .zshrc for Zsh) and that you’ve used the correct syntax.\n",
+    "- **System Restart**: Sometimes, especially on Windows, you may need to restart your system for the changes to environment variables to take effect.\n",
+    "- **Check Spelling and Syntax**: Verify that the variable `OPENAI_API_KEY` is spelled correctly in both your environment settings and your script. Also, ensure that there are no extra spaces or syntax errors in your profile scripts or environment variable settings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "\n",
+    "# Disable a few less-than-useful UserWarnings from setuptools and pydantic\n",
+    "warnings.filterwarnings(\"ignore\", category=UserWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import HTML\n",
+    "import os\n",
+    "\n",
+    "import openai\n",
+    "import pandas as pd\n",
+    "\n",
+    "import mlflow\n",
+    "from mlflow.models.signature import ModelSignature\n",
+    "from mlflow.types.schema import ColSpec, ParamSchema, ParamSpec, Schema\n",
+    "\n",
+    "\n",
+    "# Run a quick validation that we have an entry for the OPEN_API_KEY within environment variables\n",
+    "assert \"OPENAI_API_KEY\" in os.environ, \"OPENAI_API_KEY environment variable must be set\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Understanding Prompts and Their Engineering\n",
+    "\n",
+    "#### What is a Prompt?\n",
+    "A prompt is a text input given to an AI model, particularly language models like GPT-3 and GPT-4, to elicit a specific type of response or output. It guides the model on the expected information or format of the response, setting the stage for the AI's \"thought process\" and steering it toward the desired outcome.\n",
+    "\n",
+    "#### Prompt Engineering\n",
+    "Prompt engineering involves crafting these inputs to maximize the AI's response effectiveness and accuracy. It's about fine-tuning the language and structure of the prompt to align with the specific task, improving the quality and relevance of the AI's output by reducing ambiguity and directing the model's response towards the intended application.\n",
+    "\n",
+    "### A Fun and Simple Example: The Lyrics Corrector\n",
```

Comment:
Can we just demonstrate  the `tell me a {adjective} example`?

Reply:
That example is embedded in the guide documentation (the adjective prompt). I think those simple examples are great for API docs and guides, but the feedback we've been getting from users is that they want to see more complex tutorial use cases that show real-world applications of these models.